protocols/plaintext/src/lib.rs

```diff
@@ -108,144 +121,138 @@ impl UpgradeInfo for PlainText2Config {
 
 impl<C> InboundUpgrade<C> for PlainText2Config
 where
-    C: AsyncRead + AsyncWrite + Send + 'static
+    C: AsyncRead + AsyncWrite + Send + Unpin + 'static
 {
     type Output = (PeerId, PlainTextOutput<Negotiated<C>>);
     type Error = PlainTextError;
-    type Future = Box<dyn Future<Item = Self::Output, Error = Self::Error> + Send>;
+    type Future = Pin<Box<dyn Future<Output = Result<Self::Output, Self::Error>> + Send>>;
```

Comment:
```suggestion
    type Future = BoxFuture<'static, Result<Self::Output, Self::Error>>;
```

(assuming an `use futures::future::BoxFuture;`)

Reply:
Ah, cool. Wasn't aware of this type alias. Thanks.